### PR TITLE
[POND] Add IPSubnetV6 in CNS IPConfiguration

### PR DESCRIPTION
**Pull request by csfmomo**
**Reason for Change**:

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [X] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

File: tools.go.mod [hunk 1]
```diff
@@ -157,6 +157,7 @@ require (
 	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
 	github.com/modern-go/reflect2 v1.0.2 // indirect
 	github.com/moricho/tparallel v0.3.2 // indirect
+	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 // indirect
```
Comment (NihaNallappagari) [inline tools.go.mod]: Why is this change needed?

Comment (csfmomo) [reply tools.go.mod]: it's due to the lint previously and when I update the lint version, we don't need this change anymore.

File: cns/kubecontroller/nodenetworkconfig/conversion_test.go [hunk 2]
```diff
@@ -226,6 +227,35 @@ func TestCreateNCRequestFromDynamicNC(t *testing.T) {
 	}
 }
 
+func TestCreateNCRequestFromDynamicNCWithIPv6(t *testing.T) {
+	nc := v1alpha.NetworkContainer{
+		ID:                   ncID,
+		PrimaryIP:            primaryIP,
+		PrimaryIPV6:          "fd12:3456:789a::1",
```
Comment (ecigar13) [inline cns/kubecontroller/nodenetworkconfig/conversion_test.go]: nit: just ask copilot to make these variables not redundant. The IP is used in a few places.

Comment (csfmomo) [reply cns/kubecontroller/nodenetworkconfig/conversion_test.go]: updated.

File: cns/kubecontroller/nodenetworkconfig/conversion.go [hunk 3]
```diff
@@ -56,17 +56,30 @@ func CreateNCRequestFromDynamicNC(nc v1alpha.NetworkContainer) (*cns.CreateNetwo
 			NCVersion: int(nc.Version),
 		}
 	}
+	ipConfig := cns.IPConfiguration{
+		IPSubnet:         subnet,
+		GatewayIPAddress: nc.DefaultGateway,
+	}
+
+	if nc.SubnetAddressSpaceV6 != "" {
```
Comment (nairashu) [inline cns/kubecontroller/nodenetworkconfig/conversion.go]: **🔴 Bug risk: Missing `PrimaryIPV6` validation**

When `SubnetAddressSpaceV6` is set but `PrimaryIPV6` is empty, this silently creates an `IPSubnetV6` with `IPAddress: ""`. Downstream, `validateIPSubnet` in `cns/restserver/util.go` rejects empty `IPAddress`, so this would fail later in a confusing way.

The same issue exists in `CreateNCRequestFromStaticNC` below (line ~115).

Suggested fix:
```go
if nc.SubnetAddressSpaceV6 != "" {
    if nc.PrimaryIPV6 == "" {
        return nil, errors.New("PrimaryIPV6 must be set when SubnetAddressSpaceV6 is specified")
    }
    subnetV6Prefix, err := netip.ParsePrefix(nc.SubnetAddressSpaceV6)
    // ...
}
```

File: cns/kubecontroller/nodenetworkconfig/conversion_windows.go [hunk 4]
```diff
@@ -75,6 +75,7 @@ func createNCRequestFromStaticNCHelper(
 		Version:              strconv.FormatInt(nc.Version, 10), //nolint:gomnd // it's decimal
 		IPConfiguration: cns.IPConfiguration{
 			IPSubnet:         subnet,
+			IPSubnetV6:       subnetV6,
```
Comment (nairashu) [inline cns/kubecontroller/nodenetworkconfig/conversion_windows.go:78]: **🟡 `GatewayIPv6Address` inconsistency across paths**

The Linux static path (`conversion_linux.go:68`) sets `GatewayIPv6Address: nc.DefaultGatewayV6`, but neither this Windows static path nor the dynamic NC path in `conversion.go` sets it. The CRD has `DefaultGatewayV6` available in all paths.

Is this asymmetry intentional? If so, a comment explaining why would help future readers. If not, consider adding `GatewayIPv6Address: nc.DefaultGatewayV6` here and in the dynamic path for consistency.

File: cns/kubecontroller/nodenetworkconfig/conversion.go [hunk 5]
```diff
@@ -98,7 +111,19 @@ func CreateNCRequestFromStaticNC(nc v1alpha.NetworkContainer, isSwiftV2 bool, ip
 		subnet.IPAddress = primaryPrefix.Addr().String()
 	}
 
-	req, err := createNCRequestFromStaticNCHelper(nc, primaryPrefix, subnet, isSwiftV2, ipv6PrefixClamp)
+	var subnetV6 cns.IPSubnet
+	if nc.SubnetAddressSpaceV6 != "" {
```
Comment (nairashu) [inline cns/kubecontroller/nodenetworkconfig/conversion.go]: **🟢 Nit: Duplicated parsing logic**

This `SubnetAddressSpaceV6` parse-and-build block is identical to the one in `CreateNCRequestFromDynamicNC` above (line 64-73). Consider extracting a small helper:
```go
func parseIPv6Subnet(nc v1alpha.NetworkContainer) (cns.IPSubnet, error) { ... }
```
This would also centralize the `PrimaryIPV6` validation if added.

File: cns/kubecontroller/nodenetworkconfig/conversion_test.go [hunk 6]
```diff
@@ -226,6 +230,35 @@ func TestCreateNCRequestFromDynamicNC(t *testing.T) {
 	}
 }
```
Comment (nairashu) [inline cns/kubecontroller/nodenetworkconfig/conversion_test.go:276]: **🟢 Nit: Test style**

These new IPv6 tests are standalone functions. Consider adding them as subtests to the existing table-driven `TestCreateNCRequestFromDynamicNC` / `TestCreateNCRequestFromStaticNC` for consistency.

Also, there's no test for the case where `SubnetAddressSpaceV6` is empty — verifying that `IPSubnetV6` stays at its zero value. That's a useful safety net.

File: cns/kubecontroller/nodenetworkconfig/conversion.go [hunk 7]
```diff
@@ -56,17 +56,30 @@ func CreateNCRequestFromDynamicNC(nc v1alpha.NetworkContainer) (*cns.CreateNetwo
...
+		subnetV6Prefix, err := netip.ParsePrefix(nc.SubnetAddressSpaceV6)
+		if err != nil {
+			return nil, errors.Wrapf(err, "invalid SubnetAddressSpaceV6 %s", nc.SubnetAddressSpaceV6)
+		}
+		ipConfig.IPSubnetV6 = cns.IPSubnet{
+			IPAddress:    nc.PrimaryIPV6,
+			PrefixLength: uint8(subnetV6Prefix.Bits()), //#nosec G115 -- prefix bits are 0-128, fits uint8
+		}
+	}
```
[the remark this answers is not in the document]
Comment (nairashu) [reply cns/kubecontroller/nodenetworkconfig/conversion.go]: This is also a valid comment

File: cns/kubecontroller/nodenetworkconfig/conversion_windows.go (hunk 4 above)
Comment (csfmomo) [reply cns/kubecontroller/nodenetworkconfig/conversion_windows.go:78]: It's addressed in this pr
https://github.com/Azure/azure-container-networking/pull/4281

File: cns/NetworkContainerContract.go [hunk 8]
```diff
@@ -404,6 +404,7 @@ type NetworkInterfaceInfo struct {
 // IPConfiguration contains details about ip config to provision in the VM.
 type IPConfiguration struct {
 	IPSubnet           IPSubnet
+	IPSubnetV6         IPSubnet
 	DNSServers         []string
 	GatewayIPAddress   string
 	GatewayIPv6Address string
```
[the remark this answers is not in the document]
Comment (csfmomo) [reply cns/NetworkContainerContract.go]: I'll keep it as it is for consistency.

File: cns/kubecontroller/nodenetworkconfig/conversion_test.go [hunk 9]
```diff
@@ -220,18 +263,24 @@ func TestCreateNCRequestFromDynamicNC(t *testing.T) {
 				assert.Error(t, err)
 				return
 			}
-			assert.NoError(t, err)
-			assert.EqualValues(t, tt.want, got)
+			require.NoError(t, err)
+			if tt.want != nil {
+				assert.Equal(t, tt.want, got)
+			}
```
Comment (nairashu) [inline cns/kubecontroller/nodenetworkconfig/conversion_test.go:269]: **🟢 Minor observation (non-blocking)**: The new IPv6 test cases only set `wantIPSubnetV6` without `want`, so only `IPSubnetV6` is asserted — the rest of the response (e.g., `HostPrimaryIP`, `SecondaryIPConfigs`, `NCStatus`) is unchecked for these cases. This is fine since those fields are already covered by the existing test cases ("valid swift", "valid overlay", etc.) and the IPv6 tests are focused on the new field. Just noting for awareness.

**Comment by csfmomo** [comment]
azp run

**Comment by csfmomo** [comment]
azp run Azure Container Networking
